mipengine/controller/cleaner.py

```diff
@@ -147,136 +209,91 @@ def _get_node_info_by_id(self, node_id: str) -> _NodeInfoDTO:
+        cleanup_entries = []
+        for _file in self._cleanup_entries_folder_path.glob("cleanup_*.toml"):
+            try:
+                parsed_toml = toml.load(_file)
+            except Exception as exc:
+                self._logger.warning(
+                    f"Trying to read {_file.name=} raised exception: {exc}"
+                )
 
-        self._write_to_cleanup_file(parsed_toml)
+            cleanup_entries.append(_CleanupEntry(**parsed_toml))
+        return cleanup_entries
 
-    def remove_from_cleanup_file(self, context_id: str, node_id: str = None):
-        parsed_toml = self.read_cleanup_file()
-        if context_id in parsed_toml:
-            if node_id:
-                try:
-                    parsed_toml[context_id]["nodes"].remove(node_id)
-                except ValueError:
-                    self._logger.warning(
-                        f"(Cleaner::remove_from_cleanup_file) Tried to remove {node_id=} "
-                        f"for {context_id=} but this context_id.node_id is not in the "
-                        f"clean_up file.This should not happen."
-                    )
-                    pass
-            else:
-                parsed_toml.pop(context_id)
-        else:
+    def delete_file_by_context_id(self, context_id: str):
+        try:
+            self._get_file_by_context_id(context_id).unlink()
+        except FileNotFoundError as exc:
             self._logger.warning(
-                f"(Cleaner::remove_from_cleanup_file) Tried to remove {context_id=} but "
-                f"this context_id is not in the clean_up file.This should not happen."
+                f"Tried to delete file with {context_id=} but could not find such file. "
+                f"This should not happen. \n{exc=}"
             )
-            pass
+        self._logger.debug(f"Deleted file with {context_id=}")
 
-        self._write_to_cleanup_file(parsed_toml)
-
-    def read_cleanup_file(self) -> dict:
-        with self._file_lock:
-            if not os.path.isfile(self._cleanup_file_path):
+    def get_entry_by_context_id(self, context_id) -> _CleanupEntry:
+        _file = self._get_file_by_context_id(context_id)
+        if _file:
+            try:
+                parsed_toml = toml.load(_file)
+            except Exception as exc:
                 self._logger.warning(
-                    f"(Cleaner::read_cleanup_file) {self._cleanup_file_path=} does not "
-                    f"exist. This should not happen"
+                    f"Trying to read {_file.name=} raised exception: {exc}"
                 )
-                return {}
-            with open(self._cleanup_file_path, "r") as f:
+            return _CleanupEntry(**parsed_toml)
+        self._logger.warning(
+            f"Could not find entry and cleanup file with {context_id=}"
+        )
+
+    def _get_file_by_context_id(self, context_id: str):  # return type??
+        for _file in self._cleanup_entries_folder_path.glob("cleanup_*.toml"):
```

Comment:
Why not use the context if in the glob?

Reply:
there could be the following case, file 'cleanup_123456.toml' but context_id that it is looking for is '23456', so instead of doing string manipulations on the filename string, I prefer to go in the file and find the context id, since it is a tiny file.

Reply:
If there are 100 files though it will have to read each one from the disk and check if it's the one we want.

I am not sure what you mean by string manipulations but you can use:
`f"cleanup_{context_id}.toml"` and use that to get the file directly.

It's not a bug, I am just suggesting.